### PR TITLE
SOV-2613: Vesting rewards

### PR DESCRIPTION
**Pull request by tiltom**
Resolves:

-  https://sovryn.atlassian.net/browse/SOV-2613
- https://sovryn.atlassian.net/browse/SOV-2776

### CODE REVIEW

File: apps/frontend/src/app/5_pages/RewardsPage/components/Vesting/components/WithdrawButton/WithdrawButton.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+export const WithdrawButton = (item: VestingContractTableRecord) => {
+  const handleWithdraw = useHandleWithdraw(item);
+
+  return (
+    <Button
+      onClick={handleWithdraw}
+      text={t(translations.common.buttons.withdraw)}
+      type={ButtonType.button}
+      style={ButtonStyle.secondary}
```
Comment (pietro-maximoff) [inline apps/frontend/src/app/5_pages/RewardsPage/components/Vesting/components/WithdrawButton/WithdrawButton.tsx:24]: pls add a disable prop if the amount to withdraw is 0

File: apps/frontend/src/app/5_pages/RewardsPage/components/Vesting/Vesting.tsx [hunk 2]
```diff
@@ -1,7 +1,59 @@
...
+        rows={vestings}
+        noData={
+          account
+            ? t(translations.rewardPage.vesting.table.noRewards)
+            : t(translations.rewardPage.vesting.table.notConnected)
+        }
+        dataAttribute="funding-history-table"
+        rowTitle={generateRowTitle}
+      />
+      <Pagination
```
Comment (pietro-maximoff) [inline apps/frontend/src/app/5_pages/RewardsPage/components/Vesting/Vesting.tsx:49]: we should hide pagination if there are no vestings available or the user is not connected

**Review by pietro-maximoff — CHANGES_REQUESTED** [review]

File: apps/frontend/src/app/5_pages/RewardsPage/components/Vesting/Vesting.tsx (hunk 2 above)
Comment (tiltom) [reply apps/frontend/src/app/5_pages/RewardsPage/components/Vesting/Vesting.tsx:49]: We don't do it on the history page either so I wanted to stay consistent.

Comment (pietro-maximoff) [reply apps/frontend/src/app/5_pages/RewardsPage/components/Vesting/Vesting.tsx:49]: yep, makes sense

**Review by soulBit — APPROVED** [review]

**Review by pietro-maximoff — APPROVED** [review]

**Pull request merged**